seerep_srv/seerep_core_fb/src/core_fb_image.cpp

```diff
@@ -66,47 +66,49 @@ boost::uuids::uuid CoreFbImage::addData(const seerep::fb::Image& img)
   }
 }
 
-void CoreFbImage::addBoundingBoxesLabeled(const seerep::fb::BoundingBoxes2DLabeledStamped& boundingBoxes2dlabeled)
+void CoreFbImage::addLabel(const seerep::fb::DatasetUuidLabel& datasetUuidLabel)
 {
   boost::uuids::string_generator gen;
-  boost::uuids::uuid uuidMsg = gen(boundingBoxes2dlabeled.header()->uuid_msgs()->str());
-  boost::uuids::uuid uuidProject = gen(boundingBoxes2dlabeled.header()->uuid_project()->str());
+  boost::uuids::uuid uuidMsg = gen(datasetUuidLabel.datasetUuid()->str());
+  boost::uuids::uuid uuidProject = gen(datasetUuidLabel.projectUuid()->str());
 
   auto hdf5io = CoreFbGeneral::getHdf5(uuidProject, m_seerepCore, m_hdf5IoMap);
 
-  hdf5io->writeImageBoundingBox2DLabeled(boost::lexical_cast<std::string>(uuidMsg), boundingBoxes2dlabeled.labels_bb());
+  hdf5io->writeLabelsFb(boost::lexical_cast<std::string>(uuidMsg), datasetUuidLabel.datasetUuid()->str(),
+                        datasetUuidLabel.labels());
 
-  std::unordered_map<std::string, std::vector<seerep_core_msgs::LabelWithInstance>> labelWithInstancePerCategory;
-  for (auto bbWithCategory : *boundingBoxes2dlabeled.labels_bb())
+  std::unordered_map<std::string, std::vector<seerep_core_msgs::Label>> labelPerCategory;
+  for (auto labelCategory : *datasetUuidLabel.labels())
   {
-    if (bbWithCategory->boundingBox2dLabeled())
+    if (labelCategory->labels())
     {
-      std::vector<seerep_core_msgs::LabelWithInstance> labelsWithInstances;
-      for (auto bb : *bbWithCategory->boundingBox2dLabeled())
+      std::vector<seerep_core_msgs::Label> labelVector;
+      for (auto labelFb : *labelCategory->labels())
       {
-        seerep_core_msgs::LabelWithInstance labelWithInstance;
-        labelWithInstance.label = bb->labelWithInstance()->label()->label()->str();
-        labelWithInstance.labelConfidence = bb->labelWithInstance()->label()->confidence();
+        seerep_core_msgs::Label label;
+        label.label = labelFb->label()->str();
+        label.labelIdDatumaro = labelFb->labelIdDatumaro();
+        label.instanceIdDatumaro = labelFb->instanceIdDatumaro();
 
         try
         {
-          labelWithInstance.uuidInstance = gen(bb->labelWithInstance()->instanceUuid()->str());
+          label.uuidInstance = gen(labelFb->instanceUuid()->str());
         }
         catch (std::runtime_error const& e)
         {
-          labelWithInstance.uuidInstance = boost::uuids::nil_uuid();
+          label.uuidInstance = boost::uuids::nil_uuid();
         }
-        labelsWithInstances.push_back(labelWithInstance);
+        labelVector.push_back(label);
       }
-      labelWithInstancePerCategory.emplace(bbWithCategory->category()->c_str(), labelsWithInstances);
+      labelPerCategory.emplace(labelCategory->category()->c_str(), labelVector);
```

Comment:
Unnecessary implicit conversion.
```suggestion
      labelPerCategory.emplace(labelCategory->category()->str(), labelVector);
```